Now 'physics.setMKS' function can really change the b2Setting.

zero-meta opened the pull request:
1. We can tuned box2d by 'physics.setMKS' now.
2. When changed the pixels-per-meter ratio with a little bigger number, like physic.setScale(160), we will receive too much
'body shape no area' warning, And loose collision shape we defined.

zero-meta committed:
Now 'physics.setMKS' function can really change the b2Setting.

zero-meta committed:
Fixed too much body shape no area warning when change the pixels-per-meter ratio
with a litte bigger number, like physic.setScale(160).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/coronalabs/submodule-box2d?pullRequest=1) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/coronalabs/submodule-box2d?pullRequest=1) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/coronalabs/submodule-box2d?pullRequest=1) it.</sub>

zero-meta committed:
Now 'physics.setMKS' function can really change the b2Setting.

Shchvova commented:
So I was reviewing this pull request and it seems to alter how physics is solved. Would this break existing games, how would it alter physics? I am not pro in Box2D, so didn't want to dive deep if you have some quick answers.

zero-meta commented:
It will not break existing games. We fixed three problems.
1. Solar2d api 'physic.setMKS' is exposed to user before, but it has no actual effect when user call it. Now we changed the code to take the effect.
2. When we defined a fixture shape that contains points closed too much, we will receive [no area](https://github.com/coronalabs/corona/blob/master/librtt/Rtt_LuaLibPhysics.cpp#:~:text=a%20%5C%22shape%5C%22%20with-,no%20area,-%2C%20or%20nearly%20no) warning. And the shape will be ignored. But according to the [box2d official repo](https://github.com/erincatto/box2d/blob/main/src/collision/b2_polygon_shape.cpp#:~:text=((0.5f%20*%20b2_linearSlop)%20*%20(0.5f%20*%20b2_linearSlop))), the [tolerance](https://github.com/coronalabs/submodule-box2d/blob/master/Box2D/Collision/Shapes/b2PolygonShape.cpp#:~:text=0.5f%20*%20b2_linearSlop) here should be squared. We changed this to prevent ignoring of the small but normal area shapes.
3. For the 'physics.newJoint' api, we exposed a boolean param to let us control if the joined objects follow the standard collision rules or not. We make a pull request [#327](https://github.com/coronalabs/corona/pull/327) on the solar2d main repo.

Those three changes will not change the default behavior of box2d and apis defined by solar2d. So they will not break existing games.

zero-meta committed:
Add b2FakeJoint.

zero-meta committed:
Add collision filter between fixture and particles.
local particleSystem = physics.newParticleSystem
{
    filename = "particle.png",
    radius = 3,
    filter = {categoryBits=1, maskBits=0xFFFF, groupIndex=0}
}

kan6868 committed:
Optimize broadphase AABB and pair management (#2)

kan6868 committed:
Add maskbits for raycast (#9)

zero-meta committed:
fixing compilation issue.
